src/parsy/__init__.py

```diff
@@ -166,10 +166,23 @@ def combine_dict(self, combine_fn: Callable) -> Parser:
 
     def concat(self) -> Parser:
         """
-        Returns a parser that concatenates together (as a string) the previously
-        produced values.
+        Returns a parser that concatenates together the previously produced values.
+
+        This parser will join the values using the type of the input stream, so
+        when feeding bytes to the parser, the items to be joined must also be bytes.
         """
-        return self.map("".join)
+
+        @Parser
+        def parser(stream: bytes | str, index: int) -> Result:
```

Comment:
this could be `typing.AnyStr` btw. (that would retain the type if `Result` would become a generic type)